homepage/models.py

```diff
@@ -109,11 +110,39 @@ def does_course_have_prerequisites(course):
 
 
 class AppUser(models.Model):
+    # This field is the built-in django user model
+    # We currently use the following attributes from the django user model: username, password, email, is_active
     user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, primary_key=True)
 
     def __str__(self):
         return self.user.username
 
+    def toggle_user_activation(self):
+        self.user.is_active = not(self.user.is_active)
+
+    @staticmethod
+    # Creates a new appUser (without superuser permmisions) and saves it in the DB
+    # The is_active attribute of AppUser.user is automaticly set to 'True' when creating a new AppUser
+    def create_appUser(username, email, password):
+        appUser = AppUser()
+        user = User.objects.create_user(username, email, password)
+        appUser.user = user
+
+        appUser.save()
+        return appUser

```

Comment:
maybe you can create new AppUser object by:
```suggestion
def __init__(self, username, email, password):
    self.user = User.objects.create_user(username, email, password)
    self.save()
```

Reply:
> 
> 
> maybe you can create new AppUser object by:

I thought about this, but decided against it for 2 reasons :

1. I don't think the constructor of the class should save the object in the DB, this is why there is a separate method with a unique name (create_appUser) that indicates that an appUser **is created and also saved to the DB**

2. Django built-in User class also implements a static method(well not exactly but similar ) to create the user and also save it in the DB (User.objects.create_user())